File: haskellApython/expresiones.hs
--pag42
sumar:: [Int]->Int
sumar [] = 0
sumar (x:xs) = x +sumar(xs)

--pag43
invertir::Ord a=>[a]->[a]
invertir [] = []
invertir (x:xs) = (invertir xs )++[x]

--pag 44
igualLista:: Eq a => [a]->[a]->Bool
igualLista l1 l2 = l1 == l2

--pag 45
listaOrdenada:: Ord a=>[a]->Bool
listaOrdenada []=True
listaOrdenada [_]=True
listaOrdenada (x:y:xs) = (x<=y) && listaOrdenada(y:xs) 

--pag 46
mostrarUbicacion:: Ord a=>[a]->Int->a
mostrarUbicacion l n = l!!n

--pag 47
mayor::[Int]->Int
mayor (x:xs)
 | x > mayor(xs) = x
 | otherwise = mayor(xs)

--pag 48
contarPares:: [Int]->Int
contarPares []=0
contarPares lista= length [ x | x<- lista, mod x 2 ==0]

--pag 49
cuadrados::[Int]->[Int]
cuadrados [ ] = [ ]
cuadrados l = [x*x | x <-l]

--pag 50

divisible::Int->Int->Bool
divisible x y = (mod x y ) ==0

divisibles::Int->[Int]
divisibles x = [y | y <-[1..x], divisible x y]

esPrimo::Int->Bool
esPrimo n = length (divisibles n) <=2

File: haskellApython/expresiones.py

#!bin/python2.7

#pag 42
def sumar(x):
	if x==[]: return 0
	return x[0] + sumar(x[1:])

#pag 43 - Invertir lista
def invertir(listaNumeros):
    if listaNumeros == []: return listaNumeros
    return listaNumeros[::-1]

#pag 44
def igualar(x,y):
	if y!=x:
		return False
	else:
		return True
	
#pag 45 - Verificar lista ordenada
def listaOrdenada(lista):
    if lista == []: return True
    if len(lista) == 1: return True
    return (lista[0]<=lista [1] and listaOrdenada(lista[1:]))

#pag 46
def ubicacion(x,n):
	return x[n]

#pag 47 - Mayor elemento lista
def maximo(x):
    return mayor(x[0],x[1:])

def mayor(x,y):
    if y==[]: return x
    if x > y[0]: return mayor(x,y[1:])
    else: return mayor(y[0],y[1:])

#pag 48
def pares(x):
	if x==[]: return 0 
	if x[0]%2 ==0: return 1 + pares(x[1:])
	return pares(x[1:])

#pag 49 - Cuadrados de una lista
def cuadrados(lista):
    return [x**2 for x in lista]

#pag 50
def primo(x):
	if len(divisores(x)) <= 2: return True
	else: return False

def divisores(x):
	return  [ [a] for a in range(1,x+1) if x%a == 0  ]
